core/state/statedb.go

```diff
@@ -525,14 +527,18 @@ func (self *StateDB) GetRefund() uint64 {
 // Finalise finalises the state by removing the self destructed objects
 // and clears the journal as well as the refunds.
 func (s *StateDB) Finalise(deleteEmptyObjects bool) {
-	for addr := range s.stateObjectsDirty {
+
+	dirtyObjects := s.journal.flatten()
+
+	for addr, v := range dirtyObjects {
```

Comment:
Nitpick, you can replace `dirtyObjects` with `s.journal.flatten()`. Less code, cleaner.